ansibullbot/parsers/botmetadata.py

```diff
@@ -148,16 +149,35 @@ def _propagate(files, top, child, field, multivalued=True):
         def propagate_keys(data):
             files = data[u'files']
             '''maintainers and ignored keys defined at a directory level are copied to subpath'''
-            for file1, file2 in itertools.combinations(files.keys(), 2):
-                # Python 2.7 doesn't provide os.path.commonpath
-                common = os.path.commonprefix([file1, file2])
-                top = min(file1, file2)
-                child = max(file1, file2)
 
-                top_components = top.split(u'/')
-                child_components = child.split(u'/')
+            iterfiles = {}
+            filenames = sorted(list(files.keys()))
+            parent = None
+            for idp,parent in enumerate(filenames):
+
+                if parent not in iterfiles:
+                    iterfiles[parent] = []
```

Comment:
You can make iterfiles a `defaultdict`instead of using this idiom.

`
-iterfiles = {}
+iterfiles = defaultdict(list)
`
on line 153